diff --git a/dist/types/restricted/getLocale.d.ts b/dist/types/restricted/getLocale.d.ts
index 2941d2733042664c341776c7bc840ba0813994ca..0188bbd4de0cd013159a36b736ad9baf94c18c92 100644
--- a/dist/types/restricted/getLocale.d.ts
+++ b/dist/types/restricted/getLocale.d.ts
@@ -1,6 +1,6 @@
 import type { PermissionSpecificationBuilder, ValidPermissionSpecification, RestrictedMethodOptions } from '@metamask/permission-controller';
 import { PermissionType } from '@metamask/permission-controller';
-import type { NonEmptyArray } from '@metamask/utils';
+import type { Json, NonEmptyArray } from '@metamask/utils';
 import type { MethodHooksObject } from '../utils';
 declare const methodName = "snap_getLocale";
 export declare type GetLocaleMethodHooks = {
@@ -43,5 +43,5 @@ export declare const getLocaleBuilder: Readonly<{
  * @param hooks.getLocale - A function that returns the user selected locale.
  * @returns The user selected locale.
  */
-export declare function getImplementation({ getLocale }: GetLocaleMethodHooks): (_args: RestrictedMethodOptions<void>) => Promise<string>;
+export declare function getImplementation({ getLocale }: GetLocaleMethodHooks): (_args: RestrictedMethodOptions<Record<string, Json> | Json[]>) => Promise<string>;
 export {};
